### ISSUE TITLE
bug: lambda generated output

### ISSUE THREAD

**Issue by antdighex** [localstack/localstack#7194]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When invoking a lambda function, the chars in the returned JSON look aggressively escaped.
E.g. It appears that quotes are not sent plain, or even just escaped as `\"`, but they seem escaped one or two too many times, thus rendered as `\\\"`

An example of http request/response is:

```
POST /2015-03-31/functions/dc6992ae-e731-4829-a4e6-7640e4e1f21a/invocations HTTP/1.1
Host: 127.0.0.1:51537
amz-sdk-invocation-id: 098f9dfa-dfa3-9b9e-7f5e-f372b47afdfd
amz-sdk-request: ttl=20221116T230548Z;attempt=1;max=4
amz-sdk-retry: 0/0/500
Authorization: AWS4-HMAC-SHA256 Credential=accesskey/20221116/us-east-1/lambda/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;amz-sdk-retry;host;user-agent;x-amz-date;x-amz-invocation-type;x-amz-log-type, Signature=6c0597c05e092d2ce4d526e66be1cf72a85f12158abbdfb5cd180f7c99672ab7
User-Agent: aws-sdk-java/1.12.119 Mac_OS_X/12.6 OpenJDK_64-Bit_Server_VM/11.0.16.1+1-LTS java/11.0.16.1 scala/2.11.12 groovy/2.5.9 kotlin/1.3.61 vendor/BellSoft cfg/retry-mode/legacy
X-Amz-Date: 20221116T230458Z
X-Amz-Invocation-Type: RequestResponse
X-Amz-Log-Type: Tail
Transfer-Encoding: chunked
Connection: Keep-Alive


{"startTime":"2022-11-15T23:04:57.315125Z","endTime":"2022-11-16T23:04:57.316844Z","accountId":"a7e4e55f-bfef-49d4-aa5e-55af3aee6580"}


HTTP/1.1 200 
Content-Type: application/json
Content-Length: 241
Server: Werkzeug/2.1.2 Python/3.10.7
Date: Wed, 16 Nov 2022 23:05:01 GMT
X-Amz-Log-Result: G1szMm1TVEFSVCBSZXF1ZXN0SWQ6IGVlNTFlNjhhLWJlOTctMTUxOS01ZTM2LWQxNmRmNDkwMzQzMyBWZXJzaW9uOiAkTEFURVNUG1swbQobWzMybUVORCBSZXF1ZXN0SWQ6IGVlNTFlNjhhLWJlOTctMTUxOS01ZTM2LWQxNmRmNDkwMzQzMxtbMG0KG1szMm1SRVBPUlQgUmVxdWVzdElkOiBlZTUxZTY4YS1iZTk3LTE1MTktNWUzNi1kMTZkZjQ5MDM0MzMJSW5pdCBEdXJhdGlvbjogMTg4My4wMCBtcwlEdXJhdGlvbjogMTQ3Ljk0IG1zCUJpbGxlZCBEdXJhdGlvbjogMTQ4IG1zCU1lbW9yeSBTaXplOiAxNTM2IE1CCU1heCBNZW1vcnkgVXNlZDogMTg1IE1CCRtbMG0=
X-Amz-Executed-Version: $LATEST
Access-Control-Allow-Origin: *
Connection: close
Access-Control-Allow-Methods: HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH
Access-Control-Allow-Headers: authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request
Access-Control-Expose-Headers: etag,x-amz-version-id
date: Wed, 16 Nov 2022 23:05:01 GMT
server: hypercorn-h11

"\"{\\\"fileUploadResult\\\":{\\\"bucketName\\\":\\\"salesreportbucket\\\",\\\"keyName\\\":\\\"sales_report_null_2022-11-16T23:05:00.970Z.csv\\\"},\\\"remainingTimeInMillis\\\":2177, \\\"message\\\":\\\"Report generated successfully\\\"}\""
```

### Expected Behavior

A JSON string payload had better sent as-is, not escaped.
For us it is working fine when using an old version of LocalStack (localstack/localstack:0.10.4).
This unexpected behaviour occurs when switching to LocalStack 1.2.0, so one suspects that this behaviour was introduced between 0.10.4 and 1.2.0 that causes heavy output escaping.

### How are you starting LocalStack?

Custom (please describe below)

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

The LocalStack container is created by TestContainers as part of our integration tests. (TestContainers 1.16.3, for what it matters).

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

In the client code, we use `aws-java-sdk-lambda:1.12.119` to process the result of the invocation. Not sure it is relevant, though. The wire message itself seems weird to deserialize, while the client library doesn't look to play any role in this.

```java
byte[] bytes = invokeResult.getPayload().array();
String s = new String(bytes, StandardCharsets.US_ASCII);

//results in
// "\"{\\\"fileUploadResult\\\":{\\\"bucketName\\\":\\\"salesreportbucket\\\",\\\"keyName\\\":\\\"sales_report_null_2022-11-16T23:05:00.970Z.csv\\\"},\\\"remainingTimeInMillis\\\":2177, \\\"message\\\":\\\"Report generated successfully\\\"}\""
```

### Environment

```markdown
- OS: MacOs 12.6
- LocalStack: 1.2.0
```

**Comment by dominikschubert** [comment]
Hi @antdighex !

We've released our new lambda v2 implementation which is currently enabled via opt-in by setting `PROVIDER_OVERRIDE_LAMBDA=v2` when starting localstack. It is available in LocalStack starting with version `1.4` and in our `latest` images as well. We've added a bunch of improvements, also in regards to input/output data parsing/serializing. 

Please check it out and let us know if you're still experiencing the reported issue!

**Comment by antdighex** [comment]
Thanks a lot, we see it is fixed.
Sorry for the ultra-late reply, it was impossible for us to check it earlier (and we had a workaround in place).
